### ISSUE TITLE
Edge MediaStreamError on Edge ( Windows 10 Family edition )

### ISSUE THREAD

**Issue by lakano** [webrtcHacks/adapter#905]
## Versions affected

**Microsoft Edge 17.17134**

**adapter-6.4.7.js**

## Description
It's surely not a bug of adapter.js, the screen sharing feature doesn't seems to works under Windows 10 Family edition. If I test under Professional/Enterprise edition, this works.

After the authorization have been asked to the user, we click on Yes, and get the error « MediaStreamError » {constraintName: null, message: null, name: "AbortError" }

We've found a comment of Eric Lawrence on this page https://blogs.windows.com/msedgedev/2018/05/02/bringing-screen-capture-to-microsoft-edge-media-capture-api/ :
« This API requires Windows 10 Professional or Windows 10 Enterprise. »

I don't known why Microsoft do this... and we don't found any official Microsoft publication about this point. But, we would like to known if the MediaStreamError could happen in another situation, and so, if we in the case of edge and this specific error, we could conclude the user is under a Windows 10 Family edition and explain him Microsoft don't want to give this feature for this edition?

## Steps to reproduce

I can't share on a jsfiddle because of the iframe, but you can see an example here with adapter-6.4.7
https://appel.video/issue-edge-without-adapter.html

## Expected results
Detect this screen sharing feature it's not supported under the Windows 10 family edition

## Actual results
Report an MediaStreamError after the user give authorization to share his screen.

**Comment by wanecek** [comment]
**TL;DR** It's a bug in Windows, not the adapter

**Long version:**
We ran into this issue shortly after the Screen Share API was shipped in Edge (April 2018), and ended up reporting it to Microsoft ([#15960321](https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/15960321/) and [#17357055](https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/17357055/)), without receiving any meaningful response. It took us a lot of time before we were able to identify why it worked on some computers, while it mysteriously failed on others (i.e. Windows Prop vs Windows Home).

It was only when we were able to get in touch with the "partner support" (i.e. paid business support), and specifically point to the nature of the issue that we were able to get any sort of response. It turns out that there was a misconfiguration in Windows (or a communication mismatch), resulting in a required component (the `CapturePicker`, which is the native-level UI for picking screen capture source) not being included in Windows Home or Family. Woops.

After a lot of communication back and forth we were promised that the release post was to updated (which is what you linked to above, on blogs.windows.com/msedgedev), and to have a note added to the [status.microsoft.com](https://developer.microsoft.com/en-us/microsoft-edge/platform/status/screencapture/#screencapture) (but imo it's easily missed). We're still waiting for some sort of blog post to prevent others (e.g. you) having to bang their heads on this issue (preferably including a retrospective on how this bug could have slipped through their process).

As far as I can tell, and have had confirmed by the Edge support team, there is no way to detect which variant of Windows (e.g. Home, Pro, Enterprise etc) an Edge user is on from the browser. What we've had to do for our product is to parse the User-agent string and simply "block" Edge 17.XX fro being able to screen-share.  

This bug is fixed in the October release of Windows (aka Windows 1809), which has only just begun being rolled out again after being on hold for over a month due to a critical bug in Windows. 

Hope that clarifies the situation :)

**Comment by lakano** [comment]
Ahhhh ! Thank you so much @wanecek for all theses detailed explanations !

**Comment by lakano** [comment]
@wanecek  I confirm the problem is solved with the October patch, you can force the update manually with the link: https://www.microsoft.com/en-us/software-download/windows10ISO 
Now, this works on my Windows 10 Family edition ! 
Thank you again for your explanations!
